src/types/viewport.cpp

```diff
@@ -907,7 +912,7 @@ Viewport Viewport::ToOrigin() const noexcept
     if (!intersection.IsValid())
     {
         // Just put the original rectangle into the results and return early.
-        result.viewports.at(result.used++) = original;
+        result.push_back(original);

```

Comment:
The compiler also complains about this throwing exception. But logically this method does not throw anything. And I'm surprised that the `Substract` is not used anywhere. Should I make it non-`noexcept` or just catch the exception inside it?

Reply:
Oh, so this is an interesting one. `push_back` on `some` can throw an exception if it's full, so technically this one can throw an exception.
We know through static analysis (like, with our eyes) that it will never throw...

@miniksa what do you think about cases like this? This will never except until 4 becomes less than 3...

Reply:
Nnnnn. A few ideas all of which I don't really like....
1. Make it `except`: That then pollutes all callers. Yuck.
2. Catch it: Seems obnoxious to write a whole try/CATCH_LOG or try/CATCH_FAIL_FAST around it. 
3. Suppress it: A `noexcept` that throws technically goes straight to `std::terminate` which is more or less the same as try/CATCH_FAIL_FAST.
4. Add another accessor onto `til::some`. I can't think of one that wouldn't also suffer from this. If you do it with just unchecked array access, now you have a different warning and need `til::at`. If you do it with `some.at()`, that would presumably throw out of bounds too. If we add any other method, then it doesn't look quite like the STL collections it is mocking.

I'm leaning toward 2. I hate 1 and 4. I really don't like 3 as it's implicit instead of explicit like 2.
Doing 2 is also generally what we do around the codebase for this problem.

Also, this applies to `.ToOrigin()` below as well, right?